collect_app/src/main/AndroidManifest.xml

```diff
@@ -31,6 +31,11 @@
     <uses-permission android:name="android.permission.ACCESS_FINE_LOCATION" />
     <uses-permission android:name="android.permission.ACCESS_COARSE_LOCATION" />
     <uses-permission android:name="android.permission.READ_EXTERNAL_STORAGE" />
+    <!--
```

Comment:
I think this comment is the right idea but I think you may have accidentally added the wrong one! For wake lock, the explanation is here: https://groups.google.com/d/msg/firebase-talk/CXgecSxgsRE/avud-Rt7AwAJ -- basically, "without WAKE_LOCK, analytics data loss will occur."

Reading that thread and what you added in the comment suggests that the RECEIVE and C2D_MESSAGE permissions are also required, though. Is that what you're understanding as well?